react-app/src/components/schemaFormUtils.js

```diff
@@ -67,8 +70,130 @@ const uiSchema = {
   },
 };
 
+function Extractor(props) {
+  return (
+    <Accordion.Item eventKey={props.eventKey}>
+      <Accordion.Header>{props.formData.type}</Accordion.Header>
+      <Accordion.Body>
+        <p>This is a placeholder for an extractor. An input form will be added here.</p>
+      </Accordion.Body>
+    </Accordion.Item>
+  );
+}
+
+function ExtractorArray(props) {
+  const [extractors, setExtractors] = useState([]);
+  const [extractorsJSX, setExtractorsJSX] = useState([]);
+  const [types, setTypes] = useState([
+    'BaseFHIRExtractor',
+    'CSVAdverseEventExtractor',
+    'CSVCancerDiseaseStatusExtractor',
+    'CSVCancerRelatedMedicationExtractor',
+    'CSVClinicalTrialInformationExtractor',
+    'CSVConditionExtractor',
+    'CSVObservationExtractor',
+    'CSVPatientExtractor',
+    'CSVProcedureExtractor',
+    'CSVStagingExtractor',
+    'CSVTreatmentPlanChangeExtractor',
+    'Extractor',
+    'FHIRAdverseEventExtractor',
+    'FHIRAllergyIntoleranceExtractor',
+    'FHIRConditionExtractor',
+    'FHIRDocumentReferenceExtractor',
+    'FHIREncounterExtractor',
+    'FHIRMedicationOrderExtractor',
+    'FHIRMedicationRequestExtractor',
+    'FHIRMedicationStatementExtractor',
+    'FHIRObservationExtractor',
+    'FHIRPatientExtractor',
+    'FHIRProcedureExtractor',
+    'MCODERadiationProcedureExtractor',
+    'MCODESurgicalProcedureExtractor',
+  ]);
+
+  const getFormattedTypes = () =>
+    types.map((type) => (
+      <Dropdown.Item value={type} eventKey={type} key={type}>
+        {type}
+      </Dropdown.Item>
+    ));
+
+  function getDefaultExtractorObj(type = '') {
+    return {
+      label: '',
+      type,
+      constructorArgs: {
+        filePath: 'No File Chosen',
+      },
+    };
+  }
+
+  function addExtractor(eventKey) {
+    // Remove selected type from dropdown
+    const newTypes = types.filter((type) => type !== eventKey);
```

Comment:
It's not something I had ever thought about before, but is there a scenario where someone would want to add 2 extractors of the same type? It seems totally possible, but I'm not sure.

Reply:
For example: Someone has 2 CSVs of Patient data, one needs to be masked and the other doesn't, they'd have to add 2 Patient extractors with different settings, right?

Reply:
I actually think you're right here Dylan. At first blush, I thought the second extractor would just overwrite the data of the first; but if the data stored in the two CSV's are totally different this would actually work. Very interesting. 

With that said, maybe we allow for the insertion of multiple extractors of the same type.

Reply:
I changed this so that types aren't removed from the array when they're added. Additionally, types is no longer a state variable.